### PR TITLE
General fixes and improvements

### LINKED ISSUE #122: Make ocean not walkable/add kill zone in ocean

### LINKED ISSUE #204: Fix: Framerate dependent stuff !!
needs decrease
SanityController --> SumUpNeedsTick
EnemyState --> Regenerate
NPCSpawner/SanitySpawnController
(anything more?)

bare in mind, that all probabilities have to be adjusted as soon as this is implemented

### PR DESCRIPTION
**Pull request by Kaes3kuch3n**
Fixes #204 and closes #122, among other things

### CODE REVIEW

**Review by DavidPgl — DISMISSED** [review]

**Review by thuemmels — DISMISSED** [review]

**Comment by Kaes3kuch3n** [comment]
Also made ocean not walkable :)

File: Assets/Scripts/Utils/DestroyOnPlay.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using UnityEngine;
+
+namespace Utils
+{
+    public class DestroyOnPlay : MonoBehaviour
+    {
+        private void Awake()
+        {
+            Destroy(gameObject);
```
Comment (DavidPgl) [inline Assets/Scripts/Utils/DestroyOnPlay.cs:9]: Where do you use this?

**Review by DavidPgl — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Utils/DestroyOnPlay.cs (hunk 1 above)
Comment (Kaes3kuch3n) [reply Assets/Scripts/Utils/DestroyOnPlay.cs:9]: It's attached to the NotWalkableOceanPlane, which is used for generating the NavMesh

Comment (Kaes3kuch3n) [reply Assets/Scripts/Utils/DestroyOnPlay.cs:9]: But since the plane shouldn't be there in the actual game, I've added this little script

Comment (DavidPgl) [reply Assets/Scripts/Utils/DestroyOnPlay.cs:9]: I see

**Review by DavidPgl — APPROVED** [review]

**Review by thuemmels — APPROVED** [review]

**Pull request merged**